pennylane/transforms/qcut.py

```diff
@@ -364,6 +366,162 @@ def graph_to_tape(graph: MultiDiGraph) -> QuantumTape:
     return tape
 
 
+def _prep_zero_state(wire):
+    Identity(wire)
+
+
+def _prep_one_state(wire):
+    PauliX(wire)
+
+
+def _prep_plus_state(wire):
+    Hadamard(wire)
+
+
+def _prep_iplus_state(wire):
+    Hadamard(wire)
+    S(wires=wire)
+
+
+PREPARE_SETTINGS = [_prep_zero_state, _prep_one_state, _prep_plus_state, _prep_iplus_state]
+
+
+def expand_fragment_tapes(
+    tape: QuantumTape,
+) -> Tuple[List[QuantumTape], List[PrepareNode], List[MeasureNode]]:
+    """
+    Expands a fragment tape into a collection of tapes for each configuration of
+    :class:`MeasureNode` and :class:`PrepareNode` operations.
+
+    .. note::
+
+        This function is designed for use as part of the circuit cutting workflow. Check out the
+        :doc:`transforms </code/qml_transforms>` page for more details.
+
+    Args:
+        tape (QuantumTape): the fragment tape to be expanded.
+
+    Returns:
+        Tuple[List[QuantumTape], List[PrepareNode], List[MeasureNode]]: the
+        tapes corresponding to each configration, the preparation nodes and
+        the measurement nodes.
+
+    **Example**
+
+    Consider the following where ``graph`` contains a single
+    :class:`~.MeasureNode` and :class:`~.PrepareNode` within the full circuit
+    graph:
+
+    .. code-block:: python
+
+        from pennylane.transforms import qcut
+
+        >>> subgraphs, communication_graph = qcut.fragment_graph(graph)
+        >>> tapes = [qcut.graph_to_tape(sg) for sg in subgraphs]
+
+        >>> fragment_configurations = [qcut.expand_fragment_tapes(tape) for tape in tapes]
+        >>> fragment_configurations
+        [([<QuantumTape: wires=[0, 1], params=4>,
+           <QuantumTape: wires=[0, 1], params=4>,
+           <QuantumTape: wires=[0, 1], params=4>,
+           <QuantumTape: wires=[0, 1], params=4>],
+          [],
+          [MeasureNode(wires=[1])]),
+         ([<QuantumTape: wires=[1, 2], params=2>,
+           <QuantumTape: wires=[1, 2], params=2>,
+           <QuantumTape: wires=[1, 2], params=2>,
+           <QuantumTape: wires=[1, 2], params=2>],
+          [PrepareNode(wires=[1])],
+          [])]
+
```

Comment:
One option to keep it simple:
```suggestion
    Consider the following circuit, which contains a :class:`~.MeasureNode` and :class:`~.PrepareNode`
    operation:

    .. code-block: python

        from pennylane.transforms import qcut

        with qml.tape.QuantumTape() as tape:
            qcut.PrepareNode(wires=0)
            qml.RX(0.5, wires=0)
            qcut.MeasureNode(wires=0)

    We can expand over the measurement and preparation nodes using:

    >>> tapes, prep, meas = qml.transforms.expand_fragment_tapes(tape)
    >>> for t in tapes:
    ...     print(t.draw())
     0: ──I──RX(0.5)──┤ ⟨I⟩ ┤ ⟨Z⟩ 

     0: ──I──RX(0.5)──┤ ⟨X⟩ 

     0: ──I──RX(0.5)──┤ ⟨Y⟩ 
    
     0: ──X──RX(0.5)──┤ ⟨I⟩ ┤ ⟨Z⟩ 
    
     0: ──X──RX(0.5)──┤ ⟨X⟩ 
    
     0: ──X──RX(0.5)──┤ ⟨Y⟩ 
    
     0: ──H──RX(0.5)──┤ ⟨I⟩ ┤ ⟨Z⟩ 
    
     0: ──H──RX(0.5)──┤ ⟨X⟩ 
    
     0: ──H──RX(0.5)──┤ ⟨Y⟩ 
    
     0: ──H──S──RX(0.5)──┤ ⟨I⟩ ┤ ⟨Z⟩ 
    
     0: ──H──S──RX(0.5)──┤ ⟨X⟩ 
    
     0: ──H──S──RX(0.5)──┤ ⟨Y⟩ 
```